src/Illuminate/Routing/RouteCollection.php

```diff
@@ -189,7 +189,9 @@ public function match(Request $request)
      */
     protected function matchAgainstRoutes(array $routes, $request, $includingMethod = true)
     {
-        return Arr::first($routes, function ($value) use ($request, $includingMethod) {
+        return collect($routes)->sort(function ($route) {
```

Comment:
A bit of an open-ended comment: `sort` takes two parameters to compare and, iirc, ignoring the second one (by not properly returning 1 or -1) can result in what appears to be a somewhat randomized order.

Perhaps you should do `$routeA->isFallback <=> $routeB->isFallback`? I'm going to try and write a test to verify this theory (so such a change could be caught in the future)

Reply:
Spaceship!